xarray/convert.py

```diff
@@ -273,3 +274,16 @@ def from_iris(cube):
                           attrs=array_attrs, dims=dims)
     decoded_ds = decode_cf(dataarray._to_temp_dataset())
     return dataarray._from_temp_dataset(decoded_ds)
+
+
+def dataset_to_iris(dataset):
+    """ Convert a Dataset into an Iris CubeList.
+    """
+    from iris.cube import CubeList
+    return CubeList([dataset[variable].to_iris() for variable in list(dataset.data_vars)])
+
+
+def dataset_from_iris(cubelist):
+    """ Convert an Iris CubeList into a Dataset.
+    """
+    return Dataset({cube.var_name: DataArray.from_iris(cube) for cube in cubelist})
```

Comment:
Can we use the `name` attribute already on `DataArray.from_iris(cube)`? We have some special logic already for figuring out names in `DataArray.from_iris`.

Reply:
So `cube.name` instead of `cube.var_name`?

Reply:
I would say `DataArray.from_iris(cube).name`, but it probably makes sense save it in an intermediate variable to avoid converting the cube twice.